### PR TITLE
5308 log all logical imager output

### PR DESCRIPTION
**Pull request by jkho**

### CODE REVIEW

File: tools/logicalimager/tsk_logical_imager.cpp [hunk 1]
```diff
@@ -40,23 +40,72 @@ std::string GetErrorStdStr(DWORD err);
...
+
+    va_start(args, msg);
+    vsnprintf(buf, sizeof(buf), msg, args);
+    fprintf(stdout, buf);
+    // output to console file
+    fprintf(consoleFile, buf);
+    va_end(args);
+}
+
+void consoleError(const char *msg, ...) {
```
[the remark this answers is not in the document]
Comment (jkho) [reply tools/logicalimager/tsk_logical_imager.cpp]: consoleError output to stderr
consoleOutput output to stdout
Both log the output to a file.
I'll consolidate the 2 functions by adding a FILE * argument

File: tools/logicalimager/tsk_logical_imager.cpp [hunk 2]
```diff
@@ -40,23 +40,72 @@ std::string GetErrorStdStr(DWORD err);
...
+
+    va_start(args, msg);
+    vsnprintf(buf, sizeof(buf), msg, args);
+    fprintf(stderr, buf);
+    // output to console file
+    fprintf(consoleFile, buf);
+    va_end(args);
+}
+
+void logError(const char *msg, ...) {
```
[the remark this answers is not in the document]
Comment (jkho) [reply tools/logicalimager/tsk_logical_imager.cpp]: This only log the error to a file. It does not send output to the console.
I'll see if I can use a worker function for this.

File: tools/logicalimager/tsk_logical_imager.cpp [hunk 3]
```diff
@@ -40,23 +40,49 @@ std::string GetErrorStdStr(DWORD err);
...
+        pressAnyKeyToExit(code);
     }
     exit(code);
 }
 
+void openConsoleOutput(const std::string &consoleFileName) {
+    consoleFile = fopen(consoleFileName.c_str(), "w");
+    if (!consoleFile) {
+        fprintf(stderr, "ERROR: Failed to open console file %s\n", consoleFileName.c_str());
+        pressAnyKeyToExit(1);
```
Comment (APriestman) [inline tools/logicalimager/tsk_logical_imager.cpp]: I think this should call handleExit

File: tools/logicalimager/tsk_logical_imager.cpp [hunk 4]
```diff
@@ -40,23 +40,49 @@ std::string GetErrorStdStr(DWORD err);
 std::wstring GetErrorStdStrW(DWORD err);
 
 static TSK_TCHAR *progname;
-
+FILE *consoleFile = NULL;
 static std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
 
 static void pressAnyKeyToExit(int code) {
+    if (consoleFile) {
+        fclose(consoleFile);
```
Comment (APriestman) [inline tools/logicalimager/tsk_logical_imager.cpp]: The consoleFile should be closed in handleExit, since we won't always call this

**Pull request merged**